i915.c

```diff
@@ -560,16 +555,40 @@ static void *i915_bo_map(struct bo *bo, struct vma *vma, size_t plane, uint32_t
 
 		addr = (void *)(uintptr_t)gem_map.addr_ptr;
 	} else {
-
 		struct drm_i915_gem_mmap_gtt gem_map;
 		memset(&gem_map, 0, sizeof(gem_map));
 
 		gem_map.handle = bo->handles[0].u32;
 
 		ret = drmIoctl(bo->drv->fd, DRM_IOCTL_I915_GEM_MMAP_GTT, &gem_map);
 		if (ret) {
-			drv_log("DRM_IOCTL_I915_GEM_MMAP_GTT failed\n");
-			return MAP_FAILED;
+			struct driver *drv = bo->drv;
+			if ((drv->gpu_grp_type == TWO_GPU_IGPU_VIRTIO) || (drv->gpu_grp_type == THREE_GPU_IGPU_VIRTIO_DGPU)) {
```

Comment:
don't need add such check

Reply:
i think only in sriov mode, if gtt map fail need check the linear gen mmap, otherwise if gtt mmap fail, just return error.